spec/lib/miq_expression_spec.rb

```diff
@@ -488,33 +488,60 @@
       expect(sql).to eq("\"vms\".\"name\" IS NOT NULL AND \"vms\".\"name\" != ''")
     end
 
-    it "generates the SQL for a CONTAINS expression with field" do
+    it "generates the SQL for a CONTAINS expression with has_many field" do
       sql, * = MiqExpression.new("CONTAINS" => {"field" => "Vm.guest_applications-name", "value" => "foo"}).to_sql
-      expect(sql).to eq("\"vms\".\"id\" IN (SELECT DISTINCT \"guest_applications\".\"vm_or_template_id\" FROM \"guest_applications\" WHERE \"guest_applications\".\"name\" = 'foo')")
-    end
+      expected = "\"vms\".\"id\" IN (SELECT \"vms\".\"id\" FROM \"vms\" INNER JOIN \"guest_applications\" ON "\
+                 "\"guest_applications\".\"vm_or_template_id\" = \"vms\".\"id\" WHERE \"guest_applications\".\"name\" = 'foo')"
+      expect(sql).to eq(expected)
+      end
 
     it "can't generate the SQL for a CONTAINS expression with association.association-field" do
       sql, * = MiqExpression.new("CONTAINS" => {"field" => "Vm.guest_applications.host-name", "value" => "foo"}).to_sql
       expect(sql).to be_nil
     end
 
+    it "can't generate the SQL for a CONTAINS expression with belongs_to field" do
+      sql, * = MiqExpression.new("CONTAINS" => {"field" => "Vm.host-name", "value" => "foo"}).to_sql
+      expect(sql).to be_nil
+    end
+
+    it "can't generate the SQL for multi level contains with a scope" do
```

Comment:
This is not something new, I take it.  We are just adding specs to confirm this existing functionality, correct?